File: solved/10828.js
const fs = require('fs')
const input = fs.readFileSync('/dev/stdin').toString().split('\n')

const TESTCASE = input[0]

let stack = []
let temp = []
let answer = ''

for (let i = 1; i <= TESTCASE; i++) {
    temp = input[i].split(' ')
    if (temp[0] === 'push') stack[stack.length] = temp[1]
    else if (temp[0] === 'pop') {
        if (stack.length === 0) answer = answer + '-1\n'
        else answer = answer + stack.pop() + '\n'
    } else if (temp[0] === 'size') answer = answer + stack.length + '\n'
    else if (temp[0] === 'empty') {
        if (stack.length === 0) answer = answer + '1\n'
        else answer = answer + '0\n'
    } else if (temp[0] === 'top') {
        if (stack.length === 0) answer = answer + '-1\n'
        else answer = answer + stack[stack.length - 1] + '\n'
    }
}

console.log(answer)

File: Baekjoon/python/stack/10828.py
testcase = int(input())

stack = []
temp = []
answer = ''

for i in range(testcase):
    temp = input().split()
    if temp[0] == 'push':
        stack.append(temp[1])
    elif temp[0] == 'pop':
        answer += '-1\n' if len(stack) == 0 else stack.pop() + '\n'
    elif temp[0] == 'size':
        answer += str(len(stack)) + '\n'
    elif temp[0] == 'empty':
        answer += '1\n' if len(stack) == 0 else '0\n'
    elif temp[0] == 'top':
        answer += '-1\n' if len(stack) == 0 else stack[len(stack) - 1] + '\n'
print(answer)
